pkg/controllers/state/cluster.go

```diff
@@ -547,7 +547,7 @@ func (c *Cluster) newStateFromNode(ctx context.Context, node *corev1.Node, oldNo
 
 func (c *Cluster) cleanupNode(name string) {
 	if id := c.nodeNameToProviderID[name]; id != "" {
-		if c.nodes[id].NodeClaim == nil {
+		if c.nodes[id] != nil && c.nodes[id].NodeClaim == nil {
```

Comment:
I think we should understand the scenarios that this can occur under before we just ignore these cases -- it seems odd to me that we would get into this state but maybe this can happen when we have two nodes with the same providerID?

Reply:
Based on discussion in triage meeting, this could possibly be the node name instead?